mne/io/boxy/boxy.py

```diff
@@ -0,0 +1,404 @@
+            
+        ###remake montage using the transformed coordinates###
+        all_coords_trans = apply_trans(trans,all_coords)
+        all_chan_dict_trans = dict(zip(all_labels,all_coords_trans))
+        fiducial_coords_trans = apply_trans(trans,fiducial_coords)
+        
+        ###make our montage###
+        montage_trans = make_dig_montage(ch_pos=all_chan_dict_trans,coord_frame='head',
+                                                nasion = fiducial_coords_trans[0],
+                                                lpa = fiducial_coords_trans[1], 
+                                                rpa = fiducial_coords_trans[2])
+        
+        ###let's fix montage order ###
+        for i_chan in range(len(all_coords_trans)):
+            montage_trans.dig[i_chan+3]['r'] = all_coords_trans[i_chan]
+            montage_trans.ch_names[i_chan] = all_labels[i_chan]
+        req_ind = montage_trans.ch_names
+
+        # Create mne structure
+        ###create info structure###
+        info = mne.create_info(boxy_labels,srate,ch_types='fnirs_raw')
+        ###add data type and channel wavelength to info###
+        info.update(dig=montage_trans.dig, trans=trans)
+
+        # Store channel, source, and detector locations
+        # The channel location is stored in the first 3 entries of loc.
+        # The source location is stored in the second 3 entries of loc.
+        # The detector location is stored in the third 3 entries of loc.
+        # NIRx NIRSite uses MNI coordinates.
+        # Also encode the light frequency in the structure.
+
+        ###place our coordinates and wavelengths for each channel###
+        for i_chan in range(len(boxy_labels)-1):
+            temp_chn = apply_trans(trans,boxy_coords[i_chan][0:3])
+            temp_src = apply_trans(trans,boxy_coords[i_chan][3:6])
+            temp_det = apply_trans(trans,boxy_coords[i_chan][6:9])
+            temp_other = np.asarray(boxy_coords[i_chan][9:],dtype=np.float64)
+            info['chs'][i_chan]['loc'] = test = np.concatenate((temp_chn, temp_src, 
+                                                                temp_det, temp_other),axis=0)
+        info['chs'][-1]['loc'] = np.zeros((12,))        
+        raw_extras = {'source_num': source_num,
+                     'detect_num': detect_num, 
+                     'start_line': start_line,
+                     'files': files}
+
+        super(RawBOXY, self).__init__(
+            info, preload, filenames=[fname], last_samps=[last_sample],
+            raw_extras=[raw_extras], verbose=verbose)
+ 
+    def _read_segment_file(self, data, idx, fi, start, stop, cals, mult):
+        """Read a segment of data from a file.
+        """
+        source_num = self._raw_extras[fi]['source_num']
+        detect_num = self._raw_extras[fi]['detect_num']
+        start_line = self._raw_extras[fi]['start_line']
+
+        raw_data = pd.read_csv(self._raw_extras[fi]['files']['001'], skiprows=start_line, sep='\t')
```

Comment:
Eventually it would be good to use `numpy.loadtxt` or something similar instead of `pandas` for this if possible. We try not to require `Pandas` just for reading data. Also eventually if efficiency is a problem we could figure out how to read the correct subset of the CSV, but hopefully won't need to :)

Reply:
@kuziekj and I just removed the pandas requirement on here,